utils_nlp/dataset/cnndm.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script reuses some code from https://github.com/nlpyang/BertSum
+
+"""
+    Utility functions for downloading, extracting, and reading the
+    CNN/DM dataset at https://github.com/harvardnlp/sent-summary.
+    
+"""
+
+import glob
+import itertools
+import nltk
+
+nltk.download("punkt")
+from nltk import tokenize
+import os
+from os.path import isfile, join
+import sys
+import regex as re
+import torch
+import torchtext
+from torchtext.utils import download_from_url, extract_archive
+import zipfile
+
+
+from utils_nlp.dataset.url_utils import maybe_download
+from utils_nlp.models.transformers.extractive_summarization import get_dataset, get_dataloader
+
+REMAP = {"-lrb-": "(", "-rrb-": ")", "-lcb-": "{", "-rcb-": "}",
+         "-lsb-": "[", "-rsb-": "]", "``": '"', "''": '"'}
+
+def clean(x):
+    return re.sub(
+        r"-lrb-|-rrb-|-lcb-|-rcb-|-lsb-|-rsb-|``|''",
+        lambda m: REMAP.get(m.group()), x)
+
+
+def _line_iter(file_path):
+    with open(file_path, "r", encoding="utf8") as fd:
+        for line in fd:
+            yield line
+
+
+def _create_data_from_iterator(iterator, preprocessing, word_tokenizer):
+    # data = []
+    # for line in iterator:
+    #    data.append(preprocess((line, preprocessing, word_tokenizer)))
+    # return data
+    for line in iterator:
+        yield preprocess((line, preprocessing, word_tokenizer))
+
+
+def _remove_ttags(line):
+    line = re.sub(r"<t>", "", line)
+    # change </t> to <q>
+    # pyrouge test requires <q> as sentence splitter
+    line = re.sub(r"</t>", "<q>", line)
+    return line
+
+
+def _cnndm_target_sentence_tokenization(line):
+    return line.split("<q>")
+
+
+def preprocess(param):
+    """
+    Helper function to preprocess a list of paragraphs.
+
+    Args:
+        param (Tuple): params are tuple of (a list of strings, a list of preprocessing functions, and function to tokenize setences into words). A paragraph is represented with a single string with multiple setnences.
+
+    Returns:
+        list of list of strings, where each string is a token or word.
+    """
+
+    sentences, preprocess_pipeline, word_tokenize = param
+    for function in preprocess_pipeline:
+        sentences = function(sentences)
+    return [word_tokenize(sentence) for sentence in sentences]
+
+
+class Summarization(torch.utils.data.Dataset):
```

Comment:
TSDataSet would be more consistent with other datasets

Reply:
I prefer to use explicit name as I was lost when I saw TSDataset the first time.